### PR TITLE
Add "Fetch and prune all" toolstrip button

### LINKED ISSUE #4895: Feature request: save advance pull/fetch options
## Current Behavior

As a user, I am able to perform the predefined actions and additionally define the default behavior of the toolbar button.
image

## Desired Behavior

Personally, I would like to be able to have `Fetch all` to always `prune remote branches`.
I have tried to combine `Set the next selection as default` with `Open pull dialog` - unfortunately, did not the trick.

## Poposition

I don't have the perfect solution, but here are some ideas:

* Have an entry in `Commands` to prune all
* Be able to set the checkbox for prune persistently

### PR DESCRIPTION
**Pull request by zvirja**
Fixes #4895

Changes proposed in this pull request:
- Add `Fetch & prune all` menu item
- Add configurable confirmation dialog for the `Fetch & prune all` action
 
Screenshots before and after (if PR changes UI):

Toolstrip item:
image

Confirmation:
image

Setting:
image

What did I do to test the code and ensure quality:
- Performed the manual testing and covered the basic scenarios (like action run, configuration, making action as default).

Has been tested on (remove any that don't apply):
- Windows 10

------

Hi guys,

I've implemented everything we agreed on in the issue. The confirmation dialog should be enough to prevent any accidental clicks. Also it's possible to disable the confirmation if not required.

Please take a look and share your feedback 😉

### CODE REVIEW

**Comment by RussKie** [comment]
Sorry @zvirja but you'll need to rebase after #5324
I was meant to do it sooner but got caught up with other stuff. Hopefully a rebase won't be too painful.

File: GitUI/CommandsDialogs/FormBrowse.Designer.cs [hunk 1]
```diff
@@ -478,6 +480,14 @@ private void InitializeComponent()
             this.fetchAllToolStripMenuItem.Text = "Fetch all";
             this.fetchAllToolStripMenuItem.Click += new System.EventHandler(this.fetchAllToolStripMenuItem_Click);
             // 
+            // fetchPruneAllToolStripMenuItem
+            // 
+            this.fetchPruneAllToolStripMenuItem.Image = global::GitUI.Properties.Images.PullFetchPruneAll;
+            this.fetchPruneAllToolStripMenuItem.Name = "fetchPruneAllToolStripMenuItem";
+            this.fetchPruneAllToolStripMenuItem.Size = new System.Drawing.Size(239, 22);
+            this.fetchPruneAllToolStripMenuItem.Text = "Fetch && prune all";
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormBrowse.Designer.cs]: please use "and" instead of "&"

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 2]
```diff
@@ -69,9 +69,11 @@ public sealed partial class FormBrowse : GitModuleForm, IBrowseRepo
 
         private readonly TranslationString _pullFetch = new TranslationString("Pull - fetch");
         private readonly TranslationString _pullFetchAll = new TranslationString("Pull - fetch all");
+        private readonly TranslationString _pullFetchPruneAll = new TranslationString("Pull - fetch & prune all");
```
Comment (RussKie) [inline GitUI/CommandsDialogs/FormBrowse.cs]: ditto

**Comment by zvirja** [comment]
@RussKie Sorry for the belated update - was on the business trip 😖 

I rebased my code and pushed it.

Notice, I've additionally fixed the regression introduced by the #5324 - now the "Set next action as default" flag is reset after you tick the next action like it was before. I'm in doubt whether the change was intended, so decided to fix. If fix is not necessary, then I can remove the last commit.

Happy review 🥛

**Comment by gerhardol** [comment]
> Notice, I've additionally fixed the regression introduced by the #5324 - now the "Set next action as default" flag is reset after you tick the next action like it was before. I'm in doubt whether the change was intended, so decided to fix. If fix is not necessary, then I can remove the last commit.

Good to fix that - I was about to open an issue/PR myself

(or the change itself I am not enthusiastic, should have been sufficient with one dialog and one default action, not adding a fourth no-dialog alternative)

**Comment by zvirja** [comment]
> If I have set to remember my last action I want it stay that way.

Now I hopefully started to see a difference we have regarding how the feature should behave. There are two ways:
- _Always_ set last action as "default" one. Any time you pick up action from list, it becomes the default one (quickly accessible).
- One time change of the default action. How to use: `Tick the remember` -> `Tick action you want to be default`. After that the flag is reset and no matter what you select later - the favored action persists.

Previously it behaved in the second way and after your latest change it started to behave in the first way.

For me the most expected behavior is the old one. Usually you have the most popular action (like `Pull merge`) and you want to make it a default/favorite to quickly access it. So feature is exactly about that and the button label suggests this behavior. I don't see too much value in making this flag `sticky` - with git actions we don't usually have "locality", when we want to repeat the last actions for a few times...

If you did the change intentionally (and, basically, changed the feature) - then I'm fine to revert my fix regarding that and keep as is 😉 If you spotted some other issue - then please describe the scenario so I could test.

Thanks 😊

**Comment by gerhardol** [comment]
Just remember one time and then reset is really weird, but it is the wanted functionality.
I prefer the 2.x functionality.

The checkbox should be in the dialog.

**Comment by RussKie** [comment]
I now have a strong feeling we are talking about different things... 😨 
Would you be ok with removing the changes related to this discussion from the PR, and open another issue where we can discuss things?

**Comment by RussKie** [comment]
Actually @mstv has opened this issue already - #5374, let's continue there.

**Comment by zvirja** [comment]
@RussKie The reason why I included the fix here is that I thought that's an accidental regression. Now when we want to discuss that, it's indeed better to have a separate PR.

I've removed my last commit with the fix, so now the behavior is exactly how it was before.

Thanks for the review 😉

**Comment by RussKie** [comment]
It probably was an accidental regression (I need more sleep 😴) but I'd like to keep things separate.
I hope you kept the last commit so we can re-use it.

**Comment by zvirja** [comment]
@RussKie Cool to got your approval :tada: What should we do else to proceed further? :wink: Any actions from my side?

**Comment by RussKie** [comment]
> Any actions from my side?

No, I was merely giving time to other team members to voice there concerns.

**Comment by RussKie** [comment]
Thank you
Now let's head over to  #5374 and discuss the desired behaviour 😉

**Pull request merged**